packages/neon/neon/lib/src/blocs/timer.dart

```diff
@@ -0,0 +1,72 @@
+part of '../../neon.dart';
+
+abstract class TimerBlocEvents {
+  /// Register a [callback] that will be called periodically.
+  /// The time between the executions is defined by the [duration].
+  TimerBlocTimer registerTimer(final Duration duration, final VoidCallback callback);
+
+  /// Unregister a timer that has been previously registered with the bloc.
+  /// You can also use [TimerBlocTimer.cancel].
+  void unregisterTimer(final TimerBlocTimer timer);
+}
+
+abstract class TimerBlocStates {}
+
+/// Execute callbacks at defined periodic intervals.
+/// Components can register their callbacks and everything with the same periodicity will be executed at the same time.
+///
+/// The [TimerBloc] is a singleton.
+/// Sub-second timers are not supported.
+class TimerBloc extends Bloc implements TimerBlocEvents, TimerBlocStates {
```

Comment:
We've added unit tests for TimerBloc :tada:
But how do you imagine other classes implementing TimerBloc to mock this?

I'm not sure it is the best way but in the past I've done it like this:

```dart
class T {
factory T() => instance ??= T._();

@VisibleForTesting
factory T.mocked(T mock) => instance ??= mock;

T._();

@VisibleForTesting
static T? instance;
}
```

1. This let's me reset the singleton by setting it's instance to null (maybe related to dispose)
2. I can easily mock it
3. I like calling it instance rather than singleton as this helps people understand it even without knowing the singleton pattern

Reply:
Looks good. I wanted to have instances for tests so they don't interfere, but didn't think about this.